### DISCUSSION TITLE
Letzte Version für ESP32 mit 4MB Flash

### DISCUSSION THREAD

**Discussion by iukbox** [hoylabs/OpenDTU-OnBattery#1315] [Q&A]
Wo finde ich denn die letzte Version für die ESP32 mit 4MB Flash?
Ich benötige in diesem Fall die openDTU onbattery nur zur Übermittlung der Daten von 2 Victron MPPT und da ich noch einige der 'alten' ESP rumliegen habe möchte ich ungern neue kaufen.

**Comment by schlimmchen** [comment]
Du kannst auch weiterhin ESP32 mit 4MB Flash betreiben, du musst sie aber per USB aktualisieren, weil OTA-updates nicht mehr unterstützt werden.

**Comment by Manos1966** [comment]
https://opendtu-onbattery.net/firmware/

Du brauchst diese Datei:
**generic_esp32_4mb_no_ota.factory.bin**  For ESP32 with only 4 MB of flash, without support for OTA updates.

Wie wirst du dein ESP32 flashen?
Die einfachste Version waere mit dem **Community WebFlasher** dann brauchst du nicht Mal die Datei herunterzuladen, alles wird automatisch gemacht.
Infos unter: https://opendtu-onbattery.net/firmware/webinstall/

WICHTIG: Nach dem flashen, brauchst du auch ein [Device Profile](https://opendtu-onbattery.net/firmware/device_profiles/) 
Dies muss 
1. hoch-geladen werden auf deine geflashte ES32
2. ausgewaehlt/selektiert werden

**Comment by SvenSowa** [comment]
@iukbox 
Ich habe gerade die letzte Version 2024.10.22 geschrumpft. Falls du mit der eingeschränkten Funktionalität leben kannst:
https://github.com/SvenSowa/OpenDTU-OnBattery/releases/tag/2024.10.22
Die OTA Firmware ist mit 1.965.248 Bytes knapp unter dem Limit von 1.966.080 Bytes.

**Reply by schlimmchen** [reply]
> webapp gz files ultra compressed

Wie geht das von statten?

**Reply by SvenSowa** [reply]
Gzip (.gz) hat ja mehrere Kompressionslevel (1-9). Yarn scheint ein niedrigeres Level zu benutzen. Also habe ich die .gz Dateien in webapp_dist ausgepackt und mit 7-Zip gzip 9-Ultra level komprimiert. 
Ergebnis:
app.js.gz 215.32kb -> 206kb
zones.json.kb 3.95kb -> 3,77kb
Vermutlich kann man das auch in Yarn konfigurieren, dazu kenne ich mich zu wenig damit aus.
Ich hatte auch Brotli ausprobiert. Leider scheinen die Browser das nur mit https zu unterstützen.

Maximale Kompression: 211.002 bytes
7z.exe a -tgzip -mx9 -mfb=258 app.js.gz app.js

**Comment by schlimmchen** [comment]
> app.js.gz 215.32kb -> 206kb
> zones.json.kb 3.95kb -> 3,77kb

Verstehe, ich hatte es vermutet. Das lohnt sich nicht, jedenfalls nicht mehr jetzt da wir den Weg gegangen sind auf 8 MB flash zu bestehen bei OTA support.

Und nein, in nodejs/yarn scheint es nicht besser zu gehen oder anders einstellbar zu sein, jedenfalls hab ich es nicht geschafft zu recherchieren/umzusetzen.

> Ich hatte auch Brotli ausprobiert. Leider scheinen die Browser das nur mit https zu unterstützen.

Das ist völlig richtig und sehr bedauerlich. Siehe https://github.com/tbnobody/OpenDTU/pull/2037.
